pkg/v12/resource/chartoperator/current.go

```diff
@@ -87,8 +88,20 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 			return nil, microerror.Mask(err)
 		}
 
+		bytes, err := json.Marshal(releaseContent.Values)
```

Comment:
This Marshal -> Unmarshal used two times from what I see, what about create new function to it?

Reply:
This is needed to convert the map[string]interface{} we get from helm into the Values struct we use for comparison.

https://github.com/giantswarm/cluster-operator/blob/46e23fa6e95f1a4554511de0670a0ecf9680b5fa/pkg/v12/resource/chartoperator/spec.go#L24-L28

The logic is different for the current and desired state so I don't see the benefit in extracting it to a function.